src/v/kafka/server/tests/partition_reassignments_test.cc

```diff
@@ -0,0 +1,321 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/configuration.h"
+#include "kafka/protocol/alter_partition_reassignments.h"
+#include "kafka/protocol/list_partition_reassignments.h"
+#include "kafka/protocol/metadata.h"
+#include "kafka/protocol/schemata/alter_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_request.h"
+#include "kafka/server/handlers/topics/types.h"
+#include "kafka/types.h"
+#include "model/namespace.h"
+#include "redpanda/tests/fixture.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/util/defer.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+#include <optional>
+
+using namespace std::chrono_literals; // NOLINT
+
+inline ss::logger test_log("test"); // NOLINT
+
+class partition_reassignments_test_fixture : public redpanda_thread_fixture {
+public:
+    void create_topic(model::topic name, int partitions) {
+        model::topic_namespace tp_ns(model::kafka_namespace, std::move(name));
+
+        add_topic(tp_ns, partitions).get();
+
+        ss::parallel_for_each(
+          boost::irange(0, partitions),
+          [this, tp_ns](int i) {
+              return wait_for_partition_offset(
+                model::ntp(tp_ns.ns, tp_ns.tp, model::partition_id(i)),
+                model::offset(0));
+          })
+          .get();
+    }
+
+    template<typename Func>
+    auto do_with_client(Func&& f) {
+        return make_kafka_client().then(
```

Comment:
This makes a new client for every invocation of this method. This fixture can just have a reference to a client you've already setup in your unit test and assume that one is already connected. 
```
class partition_reassignments_test_fixture : public redpanda_thread_fixture {
    kafka::client::transport& _client;
...
```

Reply:
Sure I'll find a different way than creating a client on every request. I'll just pass the client in the methods that need it.